pkg/cmd/roachtest/test_runner.go

```diff
@@ -1736,56 +1745,90 @@ func (r *testRunner) teardownTest(
-	ctx context.Context, t *testImpl, testLogger *logger.Logger,
-) error {
+	ctx context.Context, t *testImpl, c *clusterImpl, testLogger *logger.Logger,
+) (inspectArtifactsErr error) {
 
 	if t.Failed() || roachtestflags.AlwaysCollectArtifacts {
 		t.L().Printf("Attempting to gather node fatal level logs for triage.")
-		out, err := gatherFatalNodeLogs(t, testLogger)
-		if err != nil {
-			return err
-		}
-		if out == "" {
+		fatalOut, gatherFatalErr := gatherFatalNodeLogs(t, testLogger)
+		if gatherFatalErr != nil {
+			// even if we encounter an error continue on, this is best effort
+			inspectArtifactsErr = errors.Join(inspectArtifactsErr, gatherFatalErr)
+		} else if fatalOut == "" {
 			t.L().Printf("No fatal level logs found.")
-			return nil
 		} else {
 			testLogger.PrintfCtx(ctx, "CockroachDB contains Fatal level logs. Up to the first 10 "+
-				"will be shown here. See node logs in artifacts for more details.\n%s", out)
-			t.appendGithubMessage(out)
-			return nil
+				"will be shown here. See node logs in artifacts for more details.\n%s", fatalOut)
+			t.appendGithubFatalLogs(fatalOut)
+		}
+
+		t.L().Printf("Attempting to gather ip node mapping")
+		ipNodeMapOut, gatherNodeIpErr := gatherNodeIpMapping(t, c)
+		if gatherNodeIpErr != nil {
+			inspectArtifactsErr = errors.Join(inspectArtifactsErr, gatherNodeIpErr)
+		} else {
+			t.appendGithubIpToNodeMapping(ipNodeMapOut)
 		}
 	}
-	return nil
+	return inspectArtifactsErr
+}
+
+// gatherNodeIpMapping attempts to gather cluster node ip information for debug
+func gatherNodeIpMapping(t *testImpl, c *clusterImpl) (string, error) {
+	var table [][]string
+	cachedCluster, err := getCachedCluster(c.name)
+	if err != nil {
+		return "", err
+	}
+	// The regex to select out this table in issues.roachtestNodeToIpRE is not
+	// strict on the number of columns i.e. columns can be added / removed
+	// without a regex change
+	table = append(table, []string{"Node", "Private IP", "Public IP", "Machine Type"})
+	for _, vmInstance := range cachedCluster.VMs {
+		table = append(table, []string{vmInstance.Name, vmInstance.PrivateIP, vmInstance.PublicIP, vmInstance.MachineType})
+	}
+	nodeIpTable, err := roachtestutil.ToMarkdownTable(table)
+	if err != nil {
+		return "", err
+	}
+	testClusterLogger, err := c.l.ChildLogger("node-ips", logger.QuietStderr, logger.QuietStdout)
+	if err != nil {
+		t.L().Printf("unable to create logger %s: %s", "node-ips", err)
+		return "", err
+	}
+	testClusterLogger.Printf("\n%s", nodeIpTable)
+	return nodeIpTable, nil
 }
 
 // gatherFatalNodeLogs attempts to gather fatal level node logs to help with
 // triage
 func gatherFatalNodeLogs(t *testImpl, testLogger *logger.Logger) (string, error) {
 	logPattern := `^F[0-9]{6}`
 	filePattern := "logs/*unredacted/cockroach*.log"
-	// *unredacted captures patterns for single node and multi-node clusters
-	// e.g. unredacted, 1.unredacted
+	// * wildcard to capture patterns for single node and multi-node clusters
```

Comment:
nit: not a big deal but some of these fatal node log fixes could've been their own commit